Fix draw_networkx_nodes return type

kbonney opened the pull request:
This PR addresses #7684 by returning the collection object rather than axis object in `draw_networkx_nodes`. Additional tests are added to check that the return types of `draw_networkx_nodes` and `draw_networkx_edges` return expected types.

MridulS commented:
This behavior was introduced in https://github.com/networkx/networkx/pull/7570, maybe @mjschwenne can review this?

dschult commented on networkx/drawing/tests/test_pylab.py:
```diff
@@ -362,6 +362,30 @@ def test_directed_edges_linestyle_sequence(style_seq):
+
+    G = nx.cubical_graph(nx.Graph)
+    dG = nx.cubical_graph(nx.DiGraph)
+    # nodes
+    nodes = nx.draw_networkx_nodes(G)
+    assert isinstance(nodes, PathCollection)
+    # edges
+    edges = nx.draw_networkx_edges(dG, arrows=True)
+    assert isinstance(edges, list)
+    if len(edges) > 0:
+        assert isinstance(edges[0], FancyArrowPatch)
+    edges = nx.draw_networkx_edges(dG, arrows=False)
+    assert isinstance(edges, LineCollection)
+    edges = nx.draw_networkx_edges(dG, arrows=None)
+    assert isinstance(edges, LineCollection)
+    edges = nx.draw_networkx_edges(G, arrows=True)
```
```suggestion
    pos = nx.spring_layout(G)
    # nodes
    nodes = nx.draw_networkx_nodes(G, pos)
    assert isinstance(nodes, PathCollection)
    # edges
    edges = nx.draw_networkx_edges(dG, pos, arrows=True)
    assert isinstance(edges, list)
    if len(edges) > 0:
        assert isinstance(edges[0], FancyArrowPatch)
    edges = nx.draw_networkx_edges(dG, pos, arrows=False)
    assert isinstance(edges, LineCollection)
    edges = nx.draw_networkx_edges(dG, pos, arrows=None)
    assert isinstance(edges, LineCollection)
    edges = nx.draw_networkx_edges(G, pos, arrows=True)
```

dschult reviewed (commented):
It looks like the original `return` was commented out with the axis returned instead.
There might be a reason, or it might have been for debugging something else and not replaced.

Meanwhile the new tests are failing, but I hope it is just a missing input arg.

mjschwenne commented:
I don't remember there being any specific reason I changed the return type on this function. I suspect that it was to help with generating images in a jupyter notebook but I'm not sure. Reverting the return type to what's promised in the doc string and what has been used historically by this function is a good idea. Technically if we wanted to keep the change (which we don't) we'd need a deprecation cycle I think that since that didn't happen the change was never met to be permanent.

kbonney commented:
Updated the tests so that they don't fail anymore.



> I don't remember there being any specific reason I changed the return type on this function. I suspect that it was to help with generating images in a jupyter notebook but I'm not sure. Reverting the return type to what's promised in the doc string and what has been used historically by this function is a good idea. Technically if we wanted to keep the change (which we don't) we'd need a deprecation cycle I think that since that didn't happen the change was never met to be permanent.

This came to my attention because some plotting functions in a software I maintain are built using `draw_networkx_nodes` and we use the collection objects to create colormaps, which breaks when the return type changes to axis.

mjschwenne reviewed (approved):
Thanks for the catch and test to make sure it doesn't happen again

dschult reviewed (approved):
Thanks @kbonney!!
And thanks @mjschwenne
